plugins/destination_testing.go

```diff
@@ -88,17 +95,48 @@ func (s *destinationTestSuite) destinationPluginTestWriteOverwrite(ctx context.C
 		return fmt.Errorf("expected 2 resource, got %d", len(resourcesRead))
 	}
 
-	if resource.Data.Equal(resourcesRead[0]) {
-		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[0])
+	expectedResourceData1 := appendSourceNameAndSyncTimeToCqTypes(sourceName, syncTime, resource1.Data)
+	expectedResourceData2 := appendSourceNameAndSyncTimeToCqTypes(sourceName, syncTime, resource2.Data)
+
+	expectedResourceData := []schema.CQTypes{
+		expectedResourceData1,
+		expectedResourceData2,
 	}
 
-	if resource2.Data.Equal(resourcesRead[1]) {
-		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[1])
+	// Because we have 2 resources, we need to sort (by UUID) to compare them effectively.
+	sort.Slice(expectedResourceData, func(i, j int) bool {
+		// We add '+2' to uuidFieldIndex because we add 2 columns (_cq_sync_time and source_name)
+		return expectedResourceData[i][uuidFieldIndex+2].String() < expectedResourceData[j][uuidFieldIndex+2].String()
+	})
+	sort.Slice(resourcesRead, func(i, j int) bool {
+		// We add '+2' to uuidFieldIndex because we add 2 columns (_cq_sync_time and source_name)
+		return resourcesRead[i][uuidFieldIndex+2].String() < resourcesRead[j][uuidFieldIndex+2].String()
+	})
+
+	if !expectedResourceData[0].Equal(resourcesRead[0]) {
+		return fmt.Errorf("expected data to be %v, got %v", expectedResourceData[0], resourcesRead[0])
```

Comment:
```suggestion
		return fmt.Errorf("expected data[0] to be %v, got %v", expectedResourceData[0], resourcesRead[0])
```

Ideally here the message should give enough context to tell the reader where it happened. Maybe something like `fmt.Errorf("expected data after first write to be %v, got %v", ...)`. Or another option is to put each section into a `t.Run` function such that they still run in order, but the error message more clearly indicates which part of the test failed.